index.js

```diff
@@ -49,6 +49,10 @@ module.exports = class Hyperswarm extends EventEmitter {
     this.explicitPeers = new Set()
     this.listening = null
 
+    this._opts = opts
+    this._root = opts._root
+    if (this._root) this._root.once('predestroy', () => this.destroy())
```

Comment:
don't use events for this, its async and needs to be awaited on destroy

Reply:
Set of sessions solved this!